query-algebrizer/src/cc.rs

```diff
@@ -83,6 +87,12 @@ impl<T> OptionEffect<T> for Option<T> {
     }
 }
 
+fn unit_type_set(t: ValueType) -> HashSet<ValueType> {
```

Comment:
I'd take a trait in `util` for making some of these easier.  Could probably do this generically across collection types with a `FromIterator` bound.